src/runtime/runtime.cs

```diff
@@ -1646,11 +1619,12 @@ internal static string GetManagedString(IntPtr op)
             if (type == PyUnicodeType)
             {
                 using var p = PyUnicode_AsUTF16String(new BorrowedReference(op));
-                int length = (int)PyUnicode_GetSize(op);
-                char* codePoints = (char*)PyBytes_AsString(p.DangerousGetAddress());
+                var bytesPtr = p.DangerousGetAddress();
+                int bytesLength = (int)Runtime.PyBytes_Size(bytesPtr);
+                char* codePoints = (char*)PyBytes_AsString(bytesPtr);
```

Comment:
NIT: better add overload for `PyBytes_AsString` that takes `BorrowedReference`. This would remove unnecessary call to `DangerousGetAddress` above.